Peaky/ReflectionExtensions.cs

```diff
@@ -0,0 +1,31 @@
+using System;
+using System.Linq;
+using System.Reflection;
+
+namespace Peaky
+{
+    internal static class ReflectionExtensions
+    {
+        public static object GetDefaultValue(this ParameterInfo parameter)
+        {
+            if (parameter.HasDefaultValue)
+            {
+                return parameter.DefaultValue;
+            }
+
+            return parameter.ParameterType.IsValueType ? Activator.CreateInstance(parameter.ParameterType) : null;
+        }
+
+        public static bool NotDefinedOn<T>(this MethodInfo method)
+        {
+            if (method.DeclaringType.GetInterface(typeof(T).Name) == null)
```

Comment:
This would be cleaner as an extension method on `Type` since the generic parameter isn't used in a strong-typed manner.